internal/terraform/node_action_trigger_plan.go

```diff
@@ -100,6 +101,19 @@ func (n *nodeActionTriggerPlanExpand) DynamicExpand(ctx EvalContext) (*Graph, tf
 		for _, key := range keys {
 			absResourceInstanceAddr := n.lifecycleActionTrigger.resourceAddress.Absolute(module).Instance(key)
 
+			if n.resourceTargets != nil {
+				matched := false
+				for _, resourceTarget := range n.resourceTargets {
+					if resourceTarget.TargetContains(absResourceInstanceAddr) {
+						matched = true
+						break
+					}
+				}
+				if !matched {
+					continue
+				}
+			}
```

Comment:
```suggestion
			if n.resourceTargets != nil {
			    // If the triggering resource was targeted, make sure the instance
			    // that triggered this was targeted specifically.
				matched := false
				for _, resourceTarget := range n.resourceTargets {
					if resourceTarget.TargetContains(absResourceInstanceAddr) {
						matched = true
						break
					}
				}
				if !matched {
					continue
				}
			}
```

just some more comments